
Forwarded: not-needed

--- a/legacy/rollup.config.js
+++ b/legacy/rollup.config.js
@@ -1,7 +1,7 @@
 import pkg from './package.json';
 import typescript from 'rollup-plugin-typescript';
-import resolve from 'rollup-plugin-node-resolve';
-import commonjs from 'rollup-plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import commonjs from '@rollup/plugin-commonjs';
 
 export default {
 	input: 'src/index.ts',
